plugins/linux-v4l2/v4l2-decoder.c

```diff
@@ -16,16 +16,28 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
 
 #include <obs-module.h>
+#include <linux/videodev2.h>
 
-#include "v4l2-mjpeg.h"
+#include "v4l2-decoder.h"
 
 #define blog(level, msg, ...) \
-	blog(level, "v4l2-input: mjpeg: " msg, ##__VA_ARGS__)
+	blog(level, "v4l2-input: decoder: " msg, ##__VA_ARGS__)
 
-int v4l2_init_mjpeg(struct v4l2_mjpeg_decoder *decoder)
+int v4l2_init_decoder(struct v4l2_decoder *decoder, int pixfmt)
 {
-	decoder->codec = avcodec_find_decoder(AV_CODEC_ID_MJPEG);
+	if (pixfmt == V4L2_PIX_FMT_MJPEG) {
+		decoder->codec = avcodec_find_decoder(AV_CODEC_ID_MJPEG);
+	}
+	if (pixfmt == V4L2_PIX_FMT_H264) {
+		decoder->codec = avcodec_find_decoder(AV_CODEC_ID_H264);
+	}
 	if (!decoder->codec) {
+		if (pixfmt == V4L2_PIX_FMT_MJPEG) {
+			blog(LOG_ERROR, "failed to find MJPEG decoder");
+		}
+		if (pixfmt == V4L2_PIX_FMT_H264) {
```

Comment:
Can these both be true at the same time? If not, then use an `else if` and link them together.